### PR TITLE
Update to Pants 2.9

### PR DESCRIPTION
**Pull request by christophermaier**
There are a few fairly mechanical changes to deal with deprecations. See each commit message for further details.

### CODE REVIEW

File: build-support/manage_virtualenv.sh [hunk 1]
```diff
@@ -26,13 +26,27 @@ function populate() {
...
 }
 
+# Retrieve a list of all Python dependencies in this project.
+#
+# Previously we used `./pants dependencies --type=3rdparty ::`, but
+# the `--type` option was removed in 2.9.0.dev0. Pants recommends this
+# implementation as an equivalent, and mentions that "in the future
+# there will be a more robust way of querying and filtering
+# dependencies." At that point, this function implementation can
+# likely be simplified.
```
Comment (wimax-grapl) [inline build-support/manage_virtualenv.sh:36]: is there a link to an issue about this? if not nbd

**Review by wimax-grapl — APPROVED** [review]

File: build-support/manage_virtualenv.sh (hunk 1 above)
Comment (christophermaier) [reply build-support/manage_virtualenv.sh:36]: I don't think so... this was basically copied from the deprecation warning I got.

**Review by jgrillo-grapl — APPROVED** [review]

File: Dockerfile.pulumi [hunk 2]
```diff
@@ -66,10 +42,7 @@ RUN pip install --no-cache-dir --upgrade pip==21.3.1 wheel==0.37.0 typing_extens
 
 COPY --chown=grapl pulumi pulumi
 
-# We need to parse the following command with `bash`, not `sh`
-SHELL ["/bin/bash", "-c"]
-
-RUN pip install --no-cache-dir -r <(PANTS_IGNORE='[".pex",".cache","venv"]' ./pants dependencies --type=3rdparty pulumi::)
+RUN pip install --no-cache-dir --requirement constraints.txt
```
Comment (wimax-grapl) [inline Dockerfile.pulumi:45]: Actually, could you:
1. move this above the COPY pulumi pulumi?
2. remove the `RUN pip install` on 41?

Comment (christophermaier) [reply Dockerfile.pulumi]: Yup

**Pull request merged**